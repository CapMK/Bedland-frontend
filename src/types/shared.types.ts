export type Getters<T extends string, K> = {
  [k in T]: K;
};

export type Building = {
  id: string;
  name: string;
  address: string;
  managerId: string;
  floors: string;
  createDate: string;
  updateDate: string;
  version: number;
};

export type Flat = {
  id: string;
  version: number;
  createDate: string;
  updateDate: string;
  buildingId: string;
  number: string;
  floor: string;
  shapePath: string;
};

export type FlatWithAddress = Flat & { address: string; city: string };

export type ResidentsFlats = {
  flatId: string;
  flatNumber: string;
  flatAddress: string;
  flatCreateDate: string;
};

export type BuildingWithCity = Building & { city: string };

export type ResidentsFlatsWithCity = ResidentsFlats & { city: string };
