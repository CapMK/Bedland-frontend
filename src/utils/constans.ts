export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
