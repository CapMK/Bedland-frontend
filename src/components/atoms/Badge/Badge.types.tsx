import { HexColor } from 'theme/theme.types';

export const BadgeVariants = ['primary', 'secondary', 'danger', 'warning'] as const;

export type BadgeVariantProps = (typeof BadgeVariants)[number];

export type BadgeVariantsType = { [key in BadgeVariantProps]: key };

export type BadgeProps = {
  variant: BadgeVariantProps;
};
export type StyledBadgePropses = {
  colors: {
    bgColor: HexColor;
    color: HexColor;
    borderColor: HexColor | 'transparent';
  };
};
