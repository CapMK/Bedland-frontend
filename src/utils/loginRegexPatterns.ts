import { InputVariant, TRegister } from 'components/atoms/Input/Input.types';
import { Getters } from 'types/shared.types';

type LoginRegexProps = Getters<InputVariant, TRegister>;

const loginRegexPattern: LoginRegexProps = {
  login: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  password: {
    pattern: /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&*(),<.>/?;:'"]).{8,24}$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
  confirmPassword: {
    required: true,
  },
};

export default loginRegexPattern;
// ^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$ <- regex for password that should work.
// /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/,
