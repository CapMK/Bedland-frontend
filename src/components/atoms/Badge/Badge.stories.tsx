import type { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import Badge from './Badge';
import { BADGE_VARIANTS } from './Badge.utils';

const meta = {
  title: 'Atoms/Badge',
  component: Badge,
  tags: ['autodocs'],
  argTypes: {
    variant: {
      options: BADGE_VARIANTS,
      control: { type: 'radio' },
      description: 'Badge variant',
      defaultValue: { summary: 'primary' },
    },
  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof Badge>;
export default meta;
type Story = StoryObj<typeof meta>;
export const Default: Story = {
  args: {
    variant: 'primary',
    children: 'Test',
  },
};
