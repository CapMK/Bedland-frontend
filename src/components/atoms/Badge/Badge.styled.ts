import styled from 'styled-components';
import { StyledBadgePropses } from './Badge.types';

export const PrimaryBadge = styled.div<StyledBadgePropses>`
  align-items: center;
  background-color: ${({ colors: { bgColor } }) => bgColor};
  border: 1px solid ${({ colors: { borderColor } }) => borderColor};
  border-radius: 8px;
  color: ${({ colors: { color } }) => color};
  display: flex;
  height: 1.5rem;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
`;

export const SecondaryBadge = styled(PrimaryBadge)<StyledBadgePropses>`
  background-color: ${({ colors: { bgColor } }) => bgColor};
  border: 1px solid ${({ colors: { borderColor } }) => borderColor};
  color: ${({ colors: { color } }) => color};
`;

export const WarningBadge = styled(PrimaryBadge)<StyledBadgePropses>`
  background-color: ${({ colors: { bgColor } }) => bgColor};
  border: 1px solid ${({ colors: { borderColor } }) => borderColor};
  color: ${({ colors: { color } }) => color};
`;

export const DangerBadge = styled(PrimaryBadge)<StyledBadgePropses>`
  background-color: 1px solid ${({ colors: { bgColor } }) => bgColor};
  border: 1px solid ${({ colors: { borderColor } }) => borderColor};
  color: ${({ colors: { color } }) => color};
`;
