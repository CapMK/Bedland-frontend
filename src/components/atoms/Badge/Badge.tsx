import { FC, PropsWithChildren } from 'react';
import { BadgeProps } from './Badge.types';
import { getStyledBadge, BADGE_VARIANTS } from './Badge.utils';

const Badge: FC<PropsWithChildren<BadgeProps>> = ({
  children,
  variant = BADGE_VARIANTS.primary,
}) => {
  const { StyledBadge, colors } = getStyledBadge(variant);

  return <StyledBadge colors={colors}>{children}</StyledBadge>;
};

export default Badge;
