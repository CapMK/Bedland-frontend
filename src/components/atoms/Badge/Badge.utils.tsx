import initialTheme from 'theme/initialTheme';
import { StyledBadgePropses, BadgeVariantsType, BadgeVariantProps } from './Badge.types';
import * as Styled from './Badge.styled';

export const BADGE_VARIANTS: BadgeVariantsType = {
  primary: 'primary',
  secondary: 'secondary',
  danger: 'danger',
  warning: 'warning',
} as const;

type BadgeStyleAndColorsValueType = {
  StyledBadge: typeof Styled.PrimaryBadge;
  colors: StyledBadgePropses['colors'];
};

type BadgeStyleAndColorsType = {
  [key in keyof BadgeVariantsType]: BadgeStyleAndColorsValueType;
};

const badgeStyleAndColors: BadgeStyleAndColorsType = {
  [BADGE_VARIANTS.primary]: {
    StyledBadge: Styled.PrimaryBadge,
    colors: {
      bgColor: initialTheme.palette.greenDark,
      borderColor: 'transparent',
      color: initialTheme.palette.white,
    },
  },
  [BADGE_VARIANTS.secondary]: {
    StyledBadge: Styled.SecondaryBadge,

    colors: {
      bgColor: initialTheme.palette.greyLight,
      borderColor: initialTheme.palette.sidebarTabs,
      color: initialTheme.palette.greyDark,
    },
  },
  [BADGE_VARIANTS.danger]: {
    StyledBadge: Styled.DangerBadge,

    colors: {
      bgColor: initialTheme.palette.redLight,
      borderColor: 'transparent',
      color: initialTheme.palette.white,
    },
  },
  [BADGE_VARIANTS.warning]: {
    StyledBadge: Styled.WarningBadge,
    colors: {
      bgColor: initialTheme.palette.yellow,
      borderColor: 'transparent',
      color: initialTheme.palette.text,
    },
  },
};

export const getStyledBadge = (
  badgeType: BadgeVariantProps = BADGE_VARIANTS.primary,
): BadgeStyleAndColorsValueType => badgeStyleAndColors[badgeType];
