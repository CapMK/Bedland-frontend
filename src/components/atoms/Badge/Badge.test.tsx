import { renderWithProviders } from 'utils/test.utils';
import { BadgeProps } from './Badge.types';
import Badge from './Badge';

describe('Badge', () => {
  const item = ({ variant }: BadgeProps) => renderWithProviders(<Badge variant={variant} />, {});

  it('should renders correctly primary badge', () => {
    expect(
      item({
        variant: 'primary',
      }),
    ).toMatchSnapshot();
  });
  it('should renders correctly secondary badge', () => {
    expect(
      item({
        variant: 'secondary',
      }),
    ).toMatchSnapshot();
  });
  it('should renders correctly danger Badge', () => {
    expect(
      item({
        variant: 'danger',
      }),
    ).toMatchSnapshot();
  });
  it('should renders correctly warning Badge', () => {
    expect(
      item({
        variant: 'warning',
      }),
    ).toMatchSnapshot();
  });
});
