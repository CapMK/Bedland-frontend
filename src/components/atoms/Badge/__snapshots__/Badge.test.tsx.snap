// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > should renders correctly danger Badge 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FF4438;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
}

.c1 {
  background-color: 1px solid #FF4438;
  border: 1px solid transparent;
  color: #fff;
}

<body>
    <div>
      <div
        class="c0 c1"
      />
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FF4438;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
}

.c1 {
  background-color: 1px solid #FF4438;
  border: 1px solid transparent;
  color: #fff;
}

<div>
    <div
      class="c0 c1"
    />
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Badge > should renders correctly primary badge 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #30BA57;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
}

<body>
    <div>
      <div
        class="c0"
      />
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #30BA57;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
}

<div>
    <div
      class="c0"
    />
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Badge > should renders correctly secondary badge 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F5F6F8;
  border: 1px solid #CACACA;
  border-radius: 8px;
  color: #767676;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
}

.c1 {
  background-color: #F5F6F8;
  border: 1px solid #CACACA;
  color: #767676;
}

<body>
    <div>
      <div
        class="c0 c1"
      />
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F5F6F8;
  border: 1px solid #CACACA;
  border-radius: 8px;
  color: #767676;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
}

.c1 {
  background-color: #F5F6F8;
  border: 1px solid #CACACA;
  color: #767676;
}

<div>
    <div
      class="c0 c1"
    />
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`Badge > should renders correctly warning Badge 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFD541;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #1B1A1A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
}

.c1 {
  background-color: #FFD541;
  border: 1px solid transparent;
  color: #1B1A1A;
}

<body>
    <div>
      <div
        class="c0 c1"
      />
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #FFD541;
  border: 1px solid transparent;
  border-radius: 8px;
  color: #1B1A1A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1rem;
  width: 100%;
}

.c1 {
  background-color: #FFD541;
  border: 1px solid transparent;
  color: #1B1A1A;
}

<div>
    <div
      class="c0 c1"
    />
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
